Install plugin-check and run it as a step in CI

lanej0 opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This installs the plugin-check plugin and executes the checks with that plugin.

Of the 5 categories of tests, only 1 passes (Accessibility) -- the other categories all have errors.  So I'm not sure what we want to do in terms of assertions and in terms of output for this.

I tried running the tests from the CLI and I tried running them in the browser (as they are here).  I like this implementation better because we can create better assertions using this -- for example, we could assert that no failures of a particular type occur.  We can also do simple assertions like "all tests passed"

Closes https://github.com/woocommerce/woocommerce/issues/47472

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Check that CI passes
2. `pnpm env:restart && pnpm test:e2e-pw activate-and-setup/plugin-check`

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

Commit:
Install plugin-check and run it as a step in CI

Commit:
Fix command in CI

Commit:
Break out test execution

Commit:
Break out test execution

Commit:
Changelog

Commit:
Tighten accessibility assertion

Commit:
Increase timeout to 120 seconds

Commit:
Take screenshots of plugin-check results

Commit:
Add a close button to dismiss store alerts (#48453)

* Add close button to store alerts

* Add changefile(s) from automation for the following project(s): woocommerce

* Fix tests

---------

Co-authored-by: github-actions <github-actions@github.com>

rjchow committed:
add: printful placement to add products task (#48520)

* add: printful placement to add products task

wavvves committed:
Add block/shortcode usage to Cart and Checkout pages in system report (#48300)

* Refactored get_pages() to reduce the amount of get_post() used, avoid unnecessary ifs, and fixed a bug where the classic shortcode block would trigger a false block usage value.

* Added block/shortcode usage to the system report for the Cart & Checkout pages. Also added a warning when a page contains both experiences.

* Added support info for templates out of sync not loading the page content.

Commit:
Add  a note in readme about connection to WooCommerce.com (#48289)

* Add  a note about connection to WooCommerce.com to readme

* Add link to the Privacy Policy

* Remove extra new lines

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

gigitux committed:
CYS: Show Patterns from PTK (#48207)

* WIP

* CYS: Refactor routing approach

* fix type

* Add changefile(s) from automation for the following project(s): woocommerce

* remove focus logic

* remove routekey prop

* fix scroll after insert pattern

* pass onNavigateBackClick callback as prop

* fix scroll

* Add changefile(s) from automation for the following project(s): woocommerce

* fix back arrow click

* fix css

* fix tooltip position

* create dedicated CSS class

* lint css

* restore change

* improve css class name

* remove not necessary class

* add github link to the todo

* lint css

* add want more pattern section

* remove not used import

* fix logic after merge trunk

* CYS: fix tooltip position

* Add changefile(s) from automation for the following project(s): woocommerce

* trigger CI

* improve logic

* use Optional Chaining

---------

Co-authored-by: github-actions <github-actions@github.com>

gigitux committed:
CYS: add pattern category in the block toolbar (#48501)

* WIP

* CYS: Refactor routing approach

* fix type

* Add changefile(s) from automation for the following project(s): woocommerce

* remove focus logic

* remove routekey prop

* fix scroll after insert pattern

* pass onNavigateBackClick callback as prop

* fix scroll

* Add changefile(s) from automation for the following project(s): woocommerce

* fix back arrow click

* fix css

* fix tooltip position

* create dedicated CSS class

* lint css

* restore change

* improve css class name

* remove not necessary class

* add github link to the todo

* lint css

* add want more pattern section

* remove not used import

* CYS: set new default patterns

* Add changefile(s) from automation for the following project(s): woocommerce

* fix logic after merge trunk

* CYS: fix tooltip position

* Add changefile(s) from automation for the following project(s): woocommerce

* trigger CI

* CYS: add pattern category in the block toolbar

* remove not necessary import

* improve style

* Add changefile(s) from automation for the following project(s): woocommerce

* improve CSS

---------

Co-authored-by: github-actions <github-actions@github.com>

ecgan committed:
Fix "Rect" to "React" (#48542)

Fix "Rect" to "React".

veljkho committed:
Fix flaky checkout coupons e2e test (#48555)

Commit:
Cherry-pick #311 into trunk (#48570)

* Redacted sensitive info from SSR for GutHub

* Redacted database section

* lint fix

---------

Co-authored-by: Naman Malhotra <naman03malhotra@gmail.com>

adimoldovan commented:
@lanej0 Thanks for picking this up!

I know it's still a draft but noticed the approach of adding e2e tests for this and I found it odd.
I think it would make more sense to use the tool in the cli and configure it as a standalone job in `config.ci.tests` in package.json. We'll probably only need it to run daily at most, or maybe even as a release check.

What do you think?

jorgeatorres committed:
Conditionally allow stock_quantity to be non-integer in REST API product schema (#48541)

albarin committed:
CYS - Optimize intro `professional-theme.svg` image (#48566)

* Optimize intro `professional-theme.svg` image

* Add changefile(s) from automation for the following project(s): woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

veljkho committed:
Fix shopper search browser sort products & merchant customer list e2e tests (#48560)

* Fix shopper search browser sort products e2e test

* Add wait for navigation

* Fix flaky merchant customer list e2e test

* Add wait for response

* Fix flaky steps

juliaamosova committed:
Update stable tag to 9.0.0 (#48584)

woorelease: Update stable tag to 9.0.0

Co-authored-by: Julia Amosova <juliaamosova@ultraw-mbp.mynetworksettings.com>

github-actions[bot] committed:
Update changelog.txt from release 9.0.0 (#48565)

Prep trunk post release 9.0.0

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

naman03malhotra committed:
Updated issue template to use the new SSR feature for GitHub (#47780)

* Update issue template to use the new SSR feature for GitHub

* Update issue template to use the new SSR feature for GitHub

* Update .github/ISSUE_TEMPLATE/1-bug-report.yml

---------

Co-authored-by: Corey McKrill <916023+coreymckrill@users.noreply.github.com>

retrofox committed:
Product Block Editor: populate attribute options optimistically  (#47618)

* show terms optimistically

* define values as TokenItem array

* introduce TokenItem type

* add token animation when adding

* terms could be undefined

* remove unused action

* changelog

* minor doc enhancements

* rename consts

* simplify items selector

* clean the temporary items, one by one

* do not pass selectedTerms when adding a new one

* pull and update form terms when creating new ones

* fix typo

* do not export TokenItem type

* rename and tweak jsdoc

* skip E2E test

chihsuan committed:
[LYS] Fix product archive page not hidden behind coming soon page (#48522)

* Fix product archive page not hidden behind the coming soon page

* Add changelog

* Update comment

* Fix tests

* Fix lint

* Fix test

chihsuan committed:
CES: Move allow tracking option call into addCustomerEffortScoreExitPageListener (#48306)

* Move fetching CES tracking logic in listener function

* Add changelog

* Remove cache

chihsuan committed:
Invalidate cache for SiteGround Speed Optimizer when coming soon options are changed (#48523)

* chore: Invalidate cache for SiteGround Speed Optimizer and WP Engine

The code changes in `ComingSoonCacheInvalidator.php` add cache invalidation for SiteGround Speed Optimizer and WP Engine. This ensures that the cache is cleared when necessary, improving the performance of the website.

* Add changelog

* Add a slash to ensure calls are made in global namespace

* Use sg_cachepress_purge_cache

* Update plugins/woocommerce/src/Internal/ComingSoon/ComingSoonCacheInvalidator.php

Co-authored-by: Adrian Duffell <9312929+adrianduffell@users.noreply.github.com>

* Fix lint

* Update plugins/woocommerce/changelog/update-purge-cache-siteground-wpengine

---------

Co-authored-by: Adrian Duffell <9312929+adrianduffell@users.noreply.github.com>
Co-authored-by: moon <moon.kyong@automattic.com>

olsonpm committed:
fix null strpos param (#48476)

* fix null strpos param

check whether $plugin_page is set prior to passing it to strpos

* Add changefile(s) from automation for the following project(s): woocommerce

* Update plugins/woocommerce/src/Internal/Admin/Orders/PageController.php

Co-authored-by: Darren Ethier <darren@roughsmootheng.in>

* Update PageController.php

fix eslint error

---------

Co-authored-by: github-actions <github-actions@github.com>
Co-authored-by: Darren Ethier <darren@roughsmootheng.in>

rjchow committed:
add/shipstation global copy (#48549)

* add: updated shipstation copy to reflect global presence

* changelogs

vedanshujain committed:
Check WC_Legacy_API instead of WC_API as that seems to be cached by lightspeed servers (#48593)

* Check WC_Legacy_API instead of WC_API as that seems to be cached by lightspeed servers.

* Improve the detection of the legacy API:

1. Modify the autoloader so that it never loads the WC_API class.

This is needed bacause the legacy REST API plugin itself does a
"class_exists('WC_API')". The plugin itself will continue to load
its own copy of the class without issues.

2. Detect the WC_Legacy_REST_API_Plugin class in the legacy API stub.

This is a performant and bulletproof way to detect if the plugin
is isntalled.

* Rearrange code a bit

---------

Co-authored-by: Nestor Soriano <konamiman@konamiman.com>

Commit:
Detect cache when launching store (#48586)

* Remove duplicate

* Add is-coming-soon-page endpoint

* Render invalid cache warning

* Use meta tag instead

* Add url for debug purpose

* Use a different copy for the congrats page when the cache is still shown

* Clean up unused codes

* Guard the api with permission

* Add changefile(s) from automation for the following project(s): woocommerce

* Reuse $is_fse_theme

* Lint fixes

* Update 48586-update-48516-detect-cache-when-launching-store

* Add changefile(s) from automation for the following project(s): woocommerce

* Add changefile(s) from automation for the following project(s): woocommerce

* Update plugins/woocommerce/changelog/48586-update-48516-detect-cache-when-launching-store

Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>

* Add changefile(s) from automation for the following project(s): woocommerce

* changed to js method

* Add changefile(s) from automation for the following project(s): @woocommerce/data, woocommerce

* empty commit to trigger CI

* Add changefile(s) from automation for the following project(s): @woocommerce/data, woocommerce

* empty commit to trigger CI

---------

Co-authored-by: github-actions <github-actions@github.com>
Co-authored-by: Chi-Hsuan Huang <chihsuan.tw@gmail.com>
Co-authored-by: rjchow <me@rjchow.com>

kmanijak committed:
Product Collection: Handpicked Products filter is broken on stores with high volume of products (#48379)

* Make the handpicked product filter in Product Collection usable for high product number stores

* Add changelog

* Include selected products in the search

* Limit the search result to 40 as 100 is unreadible anyway

* Replace fill with pressSequentially to better mimic users behaviour

* Apply two strategies for fetching products: small catalog up to 100 products and large catalog with refetching when typing

* Add comment explaining the logic

* Bring back the Loading placeholder

* Pass explicitly perPage 0 for a small catalog to fetch all the products

veljkho committed:
Fix flaky connect to Woo e2e test (#48613)

nathanss committed:
Add template URL parameter to allow choosing template directly (#48573)

mattsherman committed:
Fix terms counts in wcadmin_product_add_publish Tracks event (#48194)

* Pass changes to woocommerce_update_product

* Use woocommerce_update_product instead of wp_after_insert_post, so that terms are already updated

* Changelog

* Suppress missing hook comment phpcs error (existing hook)

* Handle woocommerce_new_product

* Make changes param optional

KIKOmanasijev committed:
Feat: add filter to toggle showing tooltips over Order statuses in da… (#47861)

* Feat: add filter to toggle showing tooltips over Order statuses in dashboard

* Use status labels when hovering over Order status badge

* wip

* Add changelog

* Fix lint issues

* Reset indentation

* Reset indentation

---------

Co-authored-by: Adrian Duffell <9312929+adrianduffell@users.noreply.github.com>

gigitux committed:
Revert "Add To Cart Form: Add `Stepper` layout" (#48624)

* Revert "Add To Cart Form: Improve the default style and add the `Stepper` style (#47664)"

This reverts commit 0712c6037b4e8a868c1e66ff18fc7046e606a23f.

* Add changefile(s) from automation for the following project(s): woocommerce-blocks, woocommerce

---------

Co-authored-by: github-actions <github-actions@github.com>

github-actions[bot] committed:
Delete changelog files based on PR 48593 (#48631)

Delete changelog files for 48593

Co-authored-by: WooCommerce Bot <no-reply@woocommerce.com>

retrofox committed:
Product Editor: show and locate help for the attribute terms component (#48600)

* show and locate the terms help

* changelog

* hide the term help once the user adds a new one

* fix typo

* always show helps for Attribute & Terms

Aljullu committed:
Update WC blocks e2e tests to WordPress 6.6 (#48436)

* Update WC blocks e2e tests to WordPress 6.6

* Add changelog file

* Small change

* Update @wordpress/e2e-test-utils-playwright package

* Fix admin.visitSiteEditor and editor.saveSiteEditorEntities

* Update pnpm-lock.yaml

* Fix template-customization.block_theme.spec.ts

* Fix Cart, Checkout and Legacy Templates tests

* Fix tests in product-gallery.block_theme.spec.ts

* Fix tests in page-content-wrapper.block_theme.spec.ts

* Fix tests in template-customization.block_theme.spec.ts

* Fix Checkout and Local pickup tests

* Make closing the welcome popup conditional

* Use editor.canvas instead of custom locator

* Remove closing the welcome popup conditional

* Update plugins/woocommerce-blocks/tests/e2e/tests/templates/template-customization.block_theme.spec.ts

Co-authored-by: Bart Kalisz <bartlomiej.kalisz@gmail.com>

* Update to WP 6.6 beta 3

* Update plugins/woocommerce-blocks/tests/e2e/tests/templates/template-customization.block_theme.spec.ts

Co-authored-by: Bart Kalisz <bartlomiej.kalisz@gmail.com>

---------

Co-authored-by: Bart Kalisz <bartlomiej.kalisz@gmail.com>

adimoldovan committed:
[test reports] Optimize the reporting job - don't duplicate reports and remove Allure report (#48612)

* * Use run_attempt in the artifact name

* Remove the Allure report steps

* Test change to trigger tests

* Only run the publish report step if there is an artifact-id

* Only run the publish report step if there is an artifact-id

* Revert test changes

* Also condition the deprecated publish report step

retrofox committed:
Product Editor: Increase the number of terms to 100 per request (#48616)

* increase items per page to 100. Initially show 20

* changelog

roykho committed:
EXPERIMENTAL: Product filters template part + overlay (#48472)

* Add overlay template part

* Add product filters overlay block

* Add changelog

* Add changefile(s) from automation for the following project(s): woocommerce-blocks, woocommerce

* Add e2e tests to ensure the filters overlay template part exists

* Remove heredoc

* Add padding to the preview

* Add inpsector controls settings

* Update plugins/woocommerce-blocks/assets/js/blocks/product-filters-overlay/editor.scss

Co-authored-by: Alexandre Lara <allexandrelara@gmail.com>

* Add position label

* Remove unneeded panelbody

---------

Co-authored-by: github-actions <github-actions@github.com>
Co-authored-by: Alexandre Lara <allexandrelara@gmail.com>

adrianduffell committed:
Update Printful copy (#48626)

* Update desination URL

* Update label copy

* Add changelog

* Revert wrong file change

* Update label copy

chihsuan committed:
Use `hasConnectedOwner` value instead to check if Jetpack is connected (#48609)

Use hasConnectedOwner value instead to check if Jetpack is connected

chihsuan committed:
Fix location settings not updated in tax task (#48606)

* Fix unable to complete tax task via wcs

* Fix missing dependencies

* Add changefile(s) from automation for the following project(s): woocommerce

* Update 48606-fix-location-not-update-in-tax-task

* Fix missing dependencies

---------

Co-authored-by: github-actions <github-actions@github.com>

psealock committed:
[Changefile] Fix Message regex (#48649)

* fix regex

* tests

Commit:
Remove e2e test, add test job for CLI

Commit:
Merge branch 'trunk' into dev/plugin-check

lanej0 commented:
This went sideways when trying to resolve a merge conflict.  Going to try again on a different branch.
